components/authz-service/server/v2/project.go

```diff
@@ -376,7 +376,7 @@ func (s *state) GetRule(ctx context.Context, req *api.GetRuleReq) (*api.GetRuleR
 			"error retrieving rule with ID %q: %s", req.Id, err.Error())
 	}
 	if resp.Deleted {
-		return nil, status.Errorf(codes.NotFound, "could not find rule with ID %q", req.Id)
+		return nil, status.Errorf(codes.NotFound, "rule with ID %q marked for deletion", req.Id)
```

Comment:
We updated the error in `UpdateRule` to be more transparent-- a rule marked for deletion is essentially "Not Found" (cannot be deleted, will not be displayed on the UI), so we backtracked and modified `CreateRule` as well.

Reply:
I wonder if we should change the shape of our restful APIs to be more clear about when an endpoint is operating on just applied rules or not (`/applied-rules` vs `/rules`). I could see getting a 404 being a bit confusing if a staged rule exists. Don't need to decide here, just something to think about 🤔